### DISCUSSION TITLE
How can I change the application settings when Ant Media Server is in cluster mode?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5161] [FAQs] [answered]
For example, to change the LiveApp settings when in standalone mode, it's possible to make changes to the `red5-web.properties` in `/usr/local/antmedia/webapps/LiveApp/WEB-INF/red5-web.properties` file, but if I try to make the same changes when in cluster mode, the changes do not apply.

**Comment by yashtandon113** [comment, marked as answer]
@timantmedia
Thanks for asking this question.

There are two ways to change application settings in cluster mode. Let me explain:

**1. Change settings directly from MongoDB database**

In first method, you can change the settings through the MongoDB database directly. You can connect to the MongoDB server and make changes using software such as Robo3T etc.
​
After gaining access to the MongoDB server, navigate to **Clusterdb**--> **App Settings**--> There will be documents on your server corresponding to the number of applications, so open and check the application name ---right click and **Edit Document** --- Change/Add the required settings and save the document.
​
![Screenshot 2023-05-30 205224](https://github.com/ant-media/Ant-Media-Server/assets/82374739/c7922cfb-c068-4ad6-9726-d7746e1a3498)

To check the string names of each setting, check https://antmedia.io/javadoc/io/antmedia/AppSettings.html

Following that, there is an **updateTime** property in the same application document. The most recent time needs to be entered there. The unix time stamp is available at https://www.epochconverter.com/.

![Screenshot 2023-05-30 205720](https://github.com/ant-media/Ant-Media-Server/assets/82374739/ee8913e7-9a9e-4cc4-a4e0-44756e8cb20d)

**2. Change settings using Rest APIs**

In this way you need to call the Management Rest APIs to change the application settings. You can use this method in standalone mode as well.

1. You need to authenticate to call management Rest APIs. You can choose any of the methods defined **[here](https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/management-rest-apis/)**.

In this example, I authenticated using user name and password.

2. Call the [Get Settings Rest API](https://antmedia.io/rest/#/ManagementRestService/getSettings) to get the settings of application. In this example, for LiveApp.

Here is the curl sample:
```
curl --location 'http://localhost:5080/rest/v2/applications/settings/LiveApp' \
--header 'Cookie: JSESSIONID=9762ACD603974026958AEAB9F0068663'
```

In response you will get all the application properties, that you need to copy and use it in next call.

3. After getting the application settings from the response of the previous Rest API call, use [Change settings Rest API](https://antmedia.io/rest/#/ManagementRestService/changeSettings) to change the settings.

For example, to disable HLS option you need to change the below property and rest other properties will be **same**.

`"hlsMuxingEnabled": false,
`
Here is the curl sample:
```
curl --location 'http://localhost:5080/rest/v2/applications/settings/LiveApp' \
--header 'Content-Type: application/json' \
--header 'Cookie: JSESSIONID=9762ACD603974026958AEAB9F0068663' \
--data '{
    "hlsMuxingEnabled": true,
}'

```
After that, you can verify the changes through the console.

**Comment by Mohit-3196** [comment]
Hi All,
Starting from Ant Media Server version 2.6.2 all the [application settings](https://antmedia.io/docs/guides/configuration-and-testing/ams-application-configuration/) can be directly set/updated from the web panel (dashboard) itself.
This is applicable for AMS running in both the Standalone as well as the Cluster mode.

To do this

- Login to the AMS web panel
- Go to the application for which you want to update the configuration and navigate to the Settings tab
- On the top right choose advanced
![Screenshot 2023-07-10 135256](https://github.com/ant-media/Ant-Media-Server/assets/86982446/91a12d9b-f8e1-4670-8bbf-eb97726af08b)

- Update the settings as needed and save it

This works for both standalone as well as cluster mode.
You can check the AMS release logs https://github.com/ant-media/Ant-Media-Server/releases
